src/liquidator.ts

```diff
@@ -1157,6 +1191,27 @@ async function closePositions(
   }
 }
 
+// Only allow strings returned by getMapKeyForOrder to be used as order placed keys
+type OrderMapKey = ReturnType<typeof getMapKeyForOrder>;
+
+function getMapKeyForOrder(
+  side: 'buy' | 'sell',
+  size: number,
+  type: 'perp' | 'spot',
+  marketIndex: number,
+) {
+  return side + size + type + marketIndex;
+}
+
+function isIdenticalOrderRecentlyPlaced(key: OrderMapKey) {
+  const now = Date.now();
+  return now - ordersPlaced.get(key) < orderDupeSafteyWindow;
```

Comment:
`ordersPlaces.get(key)` is possibly undefined, should return false in this case I guess

Reply:
Ah yeah, `now - undefined` evaluates to `NaN` and so the comparison will evaluate to false in that case but that's more of a JavaScript-y way to handle it. I'll update it to make `tsc` happy 👍